pkg/api/v1alpha1/cluster.go

```diff
@@ -303,11 +304,21 @@ func (c *Cluster) ClearPauseAnnotation() {
 }
 
 func (c *Cluster) RegistryMirror() string {
-	if c.Spec.RegistryMirrorConfiguration == nil {
+	mirrors := c.RegistryMirrors()
+	if mirrors == nil {
 		return ""
 	}
 
-	return net.JoinHostPort(c.Spec.RegistryMirrorConfiguration.Endpoint, c.Spec.RegistryMirrorConfiguration.Port)
+	return mirrors[constants.DefaultRegistry]
+}
+
+// RegistryMirrors returns registry mirror mappings of all upstream registries.
+func (c *Cluster) RegistryMirrors() map[string]string {
```

Comment:
Do we need to bubble this up to the `Cluster` object? Having 2 methods doing the same thing with very different names seems convoluted.